common/lib/xmodule/xmodule/modulestore/split_mongo/split.py

```diff
@@ -3034,11 +3065,18 @@ def _get_parents_from_structure(self, block_key, structure):
         Given a structure, find block_key's parent in that structure. Note returns
         the encoded format for parent
         """
-        return [
-            parent_block_key
-            for parent_block_key, value in structure['blocks'].iteritems()
-            if block_key in value.fields.get('children', [])
-        ]
+        cache_key = u'structure.{structure_id}'.format(structure_id=structure['_id'])
+        children_to_parents = cache.get(cache_key)
```

Comment:
yeah, even our largest structures only have a few thousand items; I can't imagine this taking more than a handful of milliseconds if we do it once